pkg/actuators/machine/machine.go

```diff
@@ -0,0 +1,321 @@
+	client client.Client,
+	machinesClient v1beta1.MachineV1beta1Interface,
+	machine *machinev1.Machine,
+	providerSpec *ovirtconfigv1.OvirtMachineProviderSpec) *machineScope {
+
+	config := ctrl.GetConfigOrDie()
+	osClient := osclientset.NewForConfigOrDie(rest.AddUserAgent(config, "cluster-api-provider-ovirt"))
+	return &machineScope{
+		Context:             ctx,
+		ovirtClient:         ovirtClient,
+		client:              client,
+		machinesClient:      machinesClient,
+		machine:             machine,
+		machineProviderSpec: providerSpec,
+		osClient:            osClient,
+	}
+}
+
+// create creates machine if it does not exists.
+func (ms *machineScope) create() error {
+	if vErr := validateMachine(ms.ovirtClient, ms.machineProviderSpec); vErr != nil {
+		return vErr
+	}
+	// creating a new instance, we don't have the vm id yet
+	instance, err := ms.ovirtClient.GetVmByName(ms.machine.Name)
+	if err != nil {
+		return err
+	}
+	// TODO: Handle that case in the actuator and make sure to return or at least check the impact on patch machine
+	if instance != nil {
+		klog.Infof("Skipped creating a VM that already exists.\n")
+		return nil
+	}
+
+	ignition, err := ms.getIgnition()
+	if err != nil {
+		return apierrors.CreateMachine("Error getting VM ignition: %v", err)
+	}
+
+	instance, err = ms.ovirtClient.CreateVmByMachine(ms.machine, ignition, ms.machineProviderSpec)
+	if err != nil {
+		return fmt.Errorf("Error creating Ovirt instance: %v", err)
+	}
+
+	// Wait till ready
+	// TODO: export to a regular function
+	// TODO: Why are we always setting error to nil?!
+	err = util.PollImmediate(RetryIntervalInstanceStatus, TimeoutInstanceCreate, func() (bool, error) {
+		instance, err := ms.ovirtClient.GetVmByMachine(*ms.machine)
+		if err != nil {
+			return false, nil
+		}
+		return instance.MustStatus() == ovirtsdk.VMSTATUS_DOWN, nil
+	})
+	if err != nil {
+		return fmt.Errorf("Error creating oVirt VM: %v", err)
+	}
+
+	// Start the VM
+	err = ms.ovirtClient.StartVM(instance.MustId())
+	if err != nil {
+		return fmt.Errorf("Error running oVirt VM: %v", err)
+	}
+
+	// Wait till running
+	// TODO: export to a regular function
+	err = util.PollImmediate(RetryIntervalInstanceStatus, TimeoutInstanceCreate, func() (bool, error) {
+		instance, err := ms.ovirtClient.GetVmByMachine(*ms.machine)
+		if err != nil {
+			return false, nil
+		}
+		return instance.MustStatus() == ovirtsdk.VMSTATUS_UP, nil
+	})
+	if err != nil {
+		return fmt.Errorf("Error running oVirt VM: %v", err)
+	}
+	return nil
+}
+
+// exists returns true if machine exists.
+func (ms *machineScope) exists() (bool, error) {
+	instance, err := ms.ovirtClient.GetVmByMachine(*ms.machine)
+	if err != nil {
+		return false, err
+	}
+	return instance != nil, err
+}
+
+func (ms *machineScope) delete() error {
+	instance, err := ms.ovirtClient.GetVmByMachine(*ms.machine)
+	if err != nil {
+		return err
+	}
+	if instance == nil {
+		klog.Infof("Skipped deleting a VM that is already deleted.\n")
```

Comment:
Consider moving klog to a dependency instead of a global function.

Reply:
Please add this to the improve loging bug